File: src/chipschallenge/blockreactions/ThiefTo.java
package chipschallenge.blockreactions;

import chipschallenge.Block;
import chipschallenge.SoundPlayer.sounds;

public class ThiefTo extends NoSlipReaction {

    private ThiefTo() {
    }
    private static ThiefTo mInstance = null;

    public static synchronized ThiefTo getInstance() {
        if (mInstance == null) {
            mInstance = new ThiefTo();
        }
        return mInstance;
    }

    public void react(Block moving, Block standing) {
        inventory().clearBoots();
        sound().playSound(sounds.THIEF);
    }

    public boolean canMove(Block moving, Block standing) {
        return moving.isChip();
    }
}

File: ChipsChallenge.Shared/Blockreactions/ThiefTo.cs
namespace ChipsChallenge.Shared.Blockreactions
{
    public class ThiefTo : NoSlipReaction
    {
        private static ThiefTo instance;

        private ThiefTo()
        {
        }

        public static ThiefTo Instance
        {
            get
            {
                lock (typeof(ThiefTo))
                {
                    if (instance == null)
                    {
                        instance = new ThiefTo();
                    }
                    return instance;
                }
            }
        }

        public override void React(Block moving, Block standing)
        {
            Inventory().ClearBoots();
            Sound().Play(Shared.Sound.Thief);
        }

        public override bool canMove(Block moving, Block standing)
        {
            return moving.Chip;
        }
    }
}